### PR TITLE
build: specify clippy denies in cargo config

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Specify clippy denies (`-D clippy::print_stdout -D clippy::print_stderr`) in cargo config. Reduce duplicate settings.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -3,3 +3,12 @@ linker = "aarch64-linux-gnu-gcc"
...
 [alias]
 sqlness = "run --bin sqlness-runner --"
+
+
+[build]
+rustflags = [
+    # lints
+    # TODO: use lint configuration in cargo https://github.com/rust-lang/cargo/issues/5034
+    "-Wclippy::print_stdout",
+    "-Wclippy::print_stderr",
```
Comment (evenyag) [inline .cargo/config.toml:13]: Should we use `-D`?

Comment (waynexia) [reply .cargo/config.toml:13]: they then `-D`'d by `-D warning`

**Review by evenyag — APPROVED** [review]

**Comment by discord9** [comment]
Maybe add a line about "-Wclippy::implicit_clone"? [clippy::implicit_clone](https://rust-lang.github.io/rust-clippy/master/#implicit_clone) deny usage of `to_vec` or `to_owned` when `clone` is available and clearer?

**Comment by waynexia** [comment]
Added in [4b20b22](https://github.com/GreptimeTeam/greptimedb/pull/1351/commits/4b20b22dc8f5e9f8d4bb03bad6cf6dafc90dd766), thanks for suggestion

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Pull request merged**